components/eamxx/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -64,6 +64,10 @@ HommeDynamics::HommeDynamics (const ekat::Comm& comm, const ekat::ParameterList&
   // Set the log filename in the F90 interface
   const char* logname = m_atm_logger->get_logfile_name().c_str();
   set_homme_log_file_name_f90 (&logname);
+
```

Comment:
I think with `params.get<int>` you can pass a default value and if the key isn't present it takes the default.  So these lines could be reduced to
```
m_bfb_has_nstep = params.get<int>("BfbHash",0);
```

Of course, if someone passes a negative number we won't catch that, so maybe add a `EKAT_ERROR_MSG(m_bfb_has_nstep>=0,..);`

Reply:
The `get` that takes a default is non-const and so can't be called on a `const PL&`. I suspect that is a typo in the PL API.

Reply:
I suppose we could allow the default on non-const PL, by simply returning the input (without setting it in the PL).

Reply:
Sure. I suggest rather than debating the nuances of PL behavior here, we accept these three lines of code as is and merge the PR.